sdk/identity/Azure.Identity/src/EnvironmentCredential.cs

```diff
@@ -49,13 +58,26 @@ public EnvironmentCredential(TokenCredentialOptions options)
             {
                 if (clientSecret != null)
                 {
-                    _credential = new ClientSecretCredential(tenantId, clientId, clientSecret, options);
+                    _credential = new ClientSecretCredential(tenantId, clientId, clientSecret, _pipeline);
                 }
                 else if (username != null && password != null && tenantId != null && clientId != null)
                 {
-                    _credential = new UsernamePasswordCredential(username, password, clientId, tenantId);
+                    _credential = new UsernamePasswordCredential(username, password, clientId, tenantId, _pipeline);
                 }
             }
+
+            if (_credential is null)
+            {
+                StringBuilder builder = new StringBuilder("EnvironmentCredential is unavailable, environment variables not fully configured.");
```

Comment:
This error makes them look as if all are required which is not true.

Reply:
updated.